### PR TITLE
Fix #398

### LINKED ISSUE #398: Can not build `oracle/tests/poseidon_tests.rs` independently
Can not build the test `oracle/tests/poseidon_tests.rs` independently. The errors say : could not find the crates : serde_json, hex, ark-serialize. 
Remove  `optional = true` part of the above dependencies in the `cargo.toml` file will work.

### PR DESCRIPTION
**Pull request by hanabi1224**
- Fixes #398 to make sure tests compile with default features
- Fix branch name in CI so that it can properly run on every push to master
- Cleanup dependencies with [cargo-udeps](https://github.com/est31/cargo-udeps) tool (Not enabled on CI tho)
- CI badge in readme.md

To further speed up the CI, I think `Check that up-to-date specification is checked in` step can be moved into a separate job that runs in parallel.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -1,9 +1,9 @@
-name: Basic rust CI stuff
+name: CI
 
 on:
   push:
     branches:
-      - main
+      - master
```
Comment (mimoo) [inline .github/workflows/rust.yml:6]: nice

File: README.md [hunk 2]
```diff
@@ -1,3 +1,6 @@
+[![CI](https://github.com/o1-labs/proof-systems/actions/workflows/rust.yml/badge.svg)](https://github.com/o1-labs/proof-systems/actions/workflows/rust.yml)
+[![dependency status](https://deps.rs/repo/github/o1-labs/proof-systems/status.svg?style=flat-square)](https://deps.rs/repo/github/o1-labs/proof-systems)
```
Comment (mimoo) [inline README.md:2]: <3

File: signer/Cargo.toml [hunk 3]
```diff
@@ -17,10 +17,12 @@ ark-ff = { version = "0.3.0", features = [ "parallel", "asm" ] }
...
-array-init = "2.0.0"
 blake2 = "0.10.0"
 hex = "0.4"
 bitvec = "1.0.0"
 sha2 = "0.10.0"
 bs58 = "0.4.0"
 thiserror = "1.0.30"
+
+[dev-dependencies]
+array-init = "2.0.0"
```
Comment (mimoo) [inline signer/Cargo.toml:28]: missing EOL

File: oracle/Cargo.toml [hunk 4]
```diff
@@ -22,12 +22,12 @@ mina-curves = { path = "../curves" }
...
-ark-serialize = { version = "0.3.0" }
+[dev-dependencies]
+serde_json = "1.0"
+hex = "0.4"
+ark-serialize = "0.3.0"
 
 [features]
 default = []
-ocaml_types = [ "ocaml", "ocaml-gen", "syn" ]
+ocaml_types = [ "ocaml", "ocaml-gen", ]
```
Comment (mimoo) [inline oracle/Cargo.toml:33]: did you use a tool to find out about all these deps?

**Review by mimoo — APPROVED** [review]
Thanks a lot! Awesome PR

> [cargo-udeps](https://github.com/est31/cargo-udeps)

I think it returns a lot of false positive right? If so not worth to have in CI

**Comment by hanabi1224** [comment]
Thanks!

> I think it returns a lot of false positive right? If so not worth to have in CI

Not really but it takes several minutes to install and run as it does not provide pre-built binaries

**Pull request merged**